internal/cloud/backend_taskStages.go

```diff
@@ -30,3 +32,127 @@ func (b *Cloud) runTaskStages(ctx context.Context, client *tfe.Client, runId str
 
 	return taskStages, nil
 }
+
+type taskStageSummarizer interface {
+	Summarize(*IntegrationContext, IntegrationOutputWriter, *tfe.TaskStage) (bool, *string, error)
```

Comment:
minor:

Probably want to add some docs about the args and outputs.. e.g. what does the first bool do in the result